tests/system_tests/test_artifacts/test_wandb_run_artifacts.py

```diff
@@ -302,46 +325,50 @@ def test_wandb_artifact_init_config(user, test_settings):
     }
 
 
-def test_log_code_settings(user):
-    with open("test.py", "w") as f:
-        f.write('print("test")')
+def test_log_code_settings(user: str, api: Api):
+    Path("test.py").write_text('print("test")')
+
     settings = wandb.Settings(save_code=True, code_dir=".")
     with wandb.init(settings=settings) as run:
         pass
 
-    artifact_name = wandb.util.make_artifact_name_safe(
+    artifact_name = make_artifact_name_safe(
         f"source-{run.project}-{run._settings.program_relpath}"
     )
-    wandb.Api().artifact(f"{artifact_name}:v0")
+    api.artifact(f"{artifact_name}:v0")
 
 
 @mark.parametrize("save_code", [True, False])
-def test_log_code_env(wandb_backend_spy, save_code):
+def test_log_code_env(
+    request: FixtureRequest, mocker: MockerFixture, wandb_backend_spy, save_code
+):
     # test for WB-7468
-    with mock.patch.dict("os.environ", WANDB_SAVE_CODE=str(save_code).lower()):
-        with open("test.py", "w") as f:
-            f.write('print("test")')
-
-        # simulate user turning on code saving in UI
-        gql = wandb_backend_spy.gql
-        wandb_backend_spy.stub_gql(
-            gql.Matcher(operation="Viewer"),
-            gql.once(
-                content={
-                    "data": {"viewer": {"flags": """{"code_saving_enabled": true}"""}}
-                },
-                status=200,
-            ),
-        )
-        settings = wandb.Settings(save_code=None, code_dir=".")
-        with wandb.init(settings=settings) as run:
-            assert run._settings.save_code is save_code
+    mocker.patch.dict(os.environ, WANDB_SAVE_CODE=str(save_code).lower())
+
+    Path("test.py").write_text('print("test")')
+
+    # simulate user turning on code saving in UI
+    gql = wandb_backend_spy.gql
+    wandb_backend_spy.stub_gql(
+        gql.Matcher(operation="Viewer"),
+        gql.once(
+            content={
+                "data": {"viewer": {"flags": """{"code_saving_enabled": true}"""}}
+            },
+            status=200,
+        ),
+    )
 
-        artifact_name = wandb.util.make_artifact_name_safe(
-            f"source-{run.project}-{run._settings.program_relpath}"
-        )
-        if save_code:
-            wandb.Api().artifact(f"{artifact_name}:v0")
-        else:
-            with raises(CommError):
-                wandb.Api().artifact(f"{artifact_name}:v0")
+    # The Api fixture must be requested AFTER the response is stubbed/mocked
+    api = request.getfixturevalue("api")
```

Comment:
This one's a little surprising to me; the `api` fixture just runs the `Api()` constructor, right? That doesn't _seem_ to request the viewer, though I could be wrong.